tests/integration/conftest.py

```diff
@@ -43,39 +40,38 @@ async def deploy(ops_test: OpsTest):
             timeout=2000,
         )
 
-    for function, alias in CHARM_FUNCTIONS.items():
-        app_name = f"superset-k8s-{alias}"
-        await ops_test.model.deploy(
-            charm,
-            resources=resources,
-            application_name=app_name,
-            config={"charm-function": function},
-            num_units=1,
-        )
+        charm = await ops_test.build_charm(".")
+        resources = {
+            "superset-image": METADATA["resources"]["superset-image"][
+                "upstream-source"
+            ]
+        }
+
+        # Iterate through UI, worker and beat charms
+        for function, alias in CHARM_FUNCTIONS.items():
+            app_name = f"superset-k8s-{alias}"
+            superset_config = {
+                "charm-function": function,
+                "superset-secret-key": "juyIKSS7cFAqJlV",
```

Comment:
You used this key twice throughout the tests. If it is important that they be a certain value, it might be better to put it as a constant somewhere.

Reply:
Good point, I have updated this.